source/code/plugins/oms_common.rb

```diff
@@ -652,7 +652,8 @@ def get_agent_version(conf_path = "/etc/opt/microsoft/omsagent/sysconf/installin
             @@AgentVersion = agent_version
           end
         end
-        return @@AgentVersion
+
+        return @@AgentVersion.nil? ? '0.0.0-0': @@AgentVersion
```

Comment:
with this approach we hit the file read every time get_agent_version is called assuming version is still not detectable, but in any case agent is a bad state. looks fine